src/main/java/gov/cabinetoffice/gap/applybackend/web/controlleradvice/ControllerExceptionHandler.java

```diff
@@ -19,12 +17,19 @@
 
 import java.time.Clock;
 import java.time.ZonedDateTime;
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 @Slf4j
 @RestControllerAdvice
 @RequiredArgsConstructor
 public class ControllerExceptionHandler {
+    public static final List<String> ADDRESS_FIELDS = List.of("addressLine1", "addressLine2", "city", "county", "postcode");
```

Comment:
Nothing wrong with this and all looks good!

Could turn this into a MAP of the fieldName and corresponding 'order number':
`public static final Map<String, Integer> ADDRESS_FIELDS_MAP = Map.of(
    "addressLine1", 1,
    "addressLine2", 2,
    "city", 3,
    "county", 4,
    "postcode", 5
);`

Then use this in place of the switch statement within '**getIntFromAddressField**' like:
`return ADDRESS_FIELDS_MAP.getOrDefault(fieldError.getFieldName(), 0);`

Seen a similar approach before with a Map, up to you whether to implement or not because the switch is also very readable!